examples/file-feed/complete/src/browser-app/DataStore.js

```diff
@@ -0,0 +1,104 @@
+ // Need to include this to make webpack happy
+import { Buffer } from 'buffer/' // eslint-disable-line no-unused-vars
+import EventEmitter from 'events'
+import spawnNode from '../util/spawn-node'
+import OrbitDB from 'orbit-db'
+import { readFileContents } from './read-file'
+
+let instance
+
+class DataStore extends EventEmitter {
+  constructor (options) {
+    super()
+    spawnNode(options, (err, node) => {
+      if (err) {
+        console.log(err)
+      }
+      this.ipfs = node
+      this.emit('ready')
+    })
+  }
+
+  // Open an orbit-db database and hook up to the emitted events
+  openFeed (name) {
+    this.orbitdb = new OrbitDB(this.ipfs)
+    this.feed = this.orbitdb.feed(name, { cachePath: '/ipfd/ipfd2.db' })
+    this.feed.events.on('ready', () => this.emit('feed'))
+    this.feed.events.on('history', () => this.emit('update'))
+    this.feed.events.on('data', () => this.emit('update'))
+    if (this.timer) clearInterval(this.timer)
+    this.timer = setInterval(() => this._updatePeers(name), 1000)
+  }
+
+  getFile (hash) {
+    return new Promise((resolve, reject) => {
```

Comment:
Back in the January meetings, we talked about using Promises in examples and we agreed at the table that we would use only callbacks for our examples. Is the use of Promises strictly necessary here?